src/services/cart.services.js

```diff
@@ -0,0 +1,57 @@
+import { redisClient } from '../helpers';
+
+async function createCart(userId) {
+  const products = await redisClient
+    .set(`cart_${userId}`, JSON.stringify([]))
+    .then(() => {
+      return true;
+    });
+  return products;
+}
+
+async function getProductsInCart(userId) {
+  const redisProducts = await redisClient.get(`cart_${userId}`);
+  if (!redisProducts) {
+    await createCart(userId);
+  }
+  const products = await redisClient.get(`cart_${userId}`);
+  const result = JSON.parse(products);
+  return result;
+}
+
+async function updateProductsInCart(userId, products) {
+  const newProducts = await redisClient
+    .set(`cart_${userId}`, JSON.stringify(products))
+    .then(() => {
+      return true;
+    });
+  return newProducts;
+}
+
+async function addToCart(userId, pid) {
+  const newCart = await getProductsInCart(userId);
+  let isInCart = false;
+  for (let i = 0; i < newCart.length; i += 1) {
+    if (newCart[i].productId === pid.productId) {
+      newCart[i].productId = pid.productId;
+      newCart[i].quantity += 1;
+      isInCart = true;
+      break;
+    }
+  }
+  if (isInCart === false) {
+    newCart.push(pid);
+  }
+  const isUpdated = await updateProductsInCart(userId, newCart).then(
```

Comment:
this syantax of async await is innvalid! plz choose either to use .then or await! 
[Read here](https://stackoverflow.com/questions/54574662/can-await-and-then-be-mixed-in-one-implementation)